src/raft.c

```diff
@@ -1656,6 +1806,10 @@ RaftReq *RaftReqInit(RedisModuleCtx *ctx, enum RaftReqType type)
         redis_raft.migrate_req = req;
     }
 
+    if (timeout > 0) {
```

Comment:
Suggest keeping `RaftReqInit()` as a constructor, and creating the timer outside of this function.

Reply:
so I didn't like that as I felt there should be symmetry between RaftReqInit and RaftReqFree.  One setups the timer, the other can tear it down.

Reply:
I think I agree with Hanna here RaftReqInit() as a constructor is good. Generally, I prefer single init function when it is possible, I think it makes reading easier when you have a single place that you set variables. 

I guess you all didn't like timer set in RaftReqInit()? I thought it was good.

Reply:
I have 2 raftreqinit's now, one with timer / one without.  the problem is, 0 is a valid timeout, so need to pass -1 to non blocking ones to distinguish them if want to do anything different.  I think it makes their calls look ugly as timeout means nothing to them.

I'm content the way it is

Reply:
ok